cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/io/cdap/cdap/datapipeline/service/ValidationHandler.java

```diff
@@ -123,12 +123,27 @@ public void validateStage(HttpServiceRequest request, HttpServiceResponder respo
     DefaultPipelineConfigurer pipelineConfigurer =
       new DefaultPipelineConfigurer(validatingConfigurer, stageConfig.getName(), Engine.SPARK, stageConfigurer);
 
+    Map<String, String> arguments = Collections.emptyMap();
+
+    // Fetch preferences for this instance and namespace and use them as program arguments if the user selects
+    // this option.
+    if (validationRequest.getResolveMacrosFromPreferences()) {
+      try {
+      arguments = getContext().getPreferencesForNamespace(namespace, true);
+      } catch (IllegalArgumentException iae) {
+        // If this method returns IllegalArgumentException, it means the namespace doesn't exist.
+        // If this is the case, we return a 404 error.
+        responder.sendError(HttpURLConnection.HTTP_NOT_FOUND,
```

Comment:
You need to `return` after this statement to terminate the flow.

Reply:
Oops, great catch.